source/common/runtime/runtime_impl.cc

```diff
@@ -205,23 +205,26 @@ bool SnapshotImpl::featureEnabled(const std::string& key,
                                   const envoy::type::FractionalPercent& default_value,
                                   uint64_t random_value) const {
   const auto& entry = values_.find(key);
-  uint64_t numerator, denominator;
+  envoy::type::FractionalPercent percent;
   if (entry != values_.end() && entry->second.fractional_percent_value_.has_value()) {
-    numerator = entry->second.fractional_percent_value_->numerator();
-    denominator = ProtobufPercentHelper::fractionalPercentDenominatorToInt(
-        entry->second.fractional_percent_value_->denominator());
+    percent = entry->second.fractional_percent_value_.value();
   } else if (entry != values_.end() && entry->second.uint_value_.has_value()) {
-    // The runtime value must have been specified as an integer rather than a fractional percent
-    // proto. To preserve legacy semantics, we'll assume this represents a percentage.
-    numerator = entry->second.uint_value_.value();
-    denominator = 100;
+    // The runtime value must have been specified as an integer rather than a
+    // fractional percent proto. To preserve legacy semantics, we'll assume
+    // this represents a percentage (i.e. denominator of 100).
+    if (entry->second.uint_value_.value() > 100) {
```

Comment:
this is maybe a bit odd? I did it because it's a uint64 and FractionalPercent uses uint32 for the numerator

Reply:
I would move the comments around. Comment above the if-statement the rationale for it and move the comment above down to be above where it's being set.

Reply:
Also, it is a bit weird, but understandable once the comments go above it.

Reply:
rearranged the comments and also changed their wording, lmk what you think